assets/js/signup.js

```diff
@@ -90,13 +113,18 @@ var questionRules = function(question) {
     rule["required"] = true;
   }
   if (question.validationType != null) {
+    // There should really be some form of validation for the validationType
+    // One option would be to make a new table with all of the possible validation types
+    // which would be both a foreign key for the validationType field in the questionInformation table
+    // and a way to validate at this point too. If the validationType has a typo for any
+    // given field, the front-end form validation fails to run at all for any of the fields.
```

Comment:
Why does one typo cause all the rest to fail? Can't you just like wrap each one in a try-catch or something..?

I suppose I'm not opposed to adding a ValidationType table (or maybe it could just be an enum, that would solve your problem too), however, it does seem a little over-engineered.. How many of the fields are not unique validations? Like phoneUS will probably only be used once... something like email would only be used once.. I don't see many instances where there would be repeat validationTypes on Questions, so I wouldn't want to add something like this. Although, I may consider the enum option, I just don't know how nicely this would play with evolving the database when we add a new validationType..